validator/src/main/java/com/amazon/aoc/validators/CWMetricValidator.java

```diff
@@ -90,29 +92,47 @@ public void validate() throws Exception {
+          // We will query metrics with specific [Service], [RemoteService], or [RemoteService, RemoteTarget] dimension
+          // values to ensure we get all metrics from all aggregations, specifically the [RemoteService] aggregation.
           List<String> serviceNames =
               Lists.newArrayList(
                   context.getServiceName(), context.getRemoteServiceDeploymentName());
-          // TODO - Put the following back in: "www.amazon.com",  "AWS.SDK.S3"
+          // TODO - Put the following back in: "www.amazon.com"
           List<String> remoteServiceNames =
               Lists.newArrayList(context.getRemoteServiceDeploymentName());
           if (context.getRemoteServiceName() != null && !context.getRemoteServiceName().isEmpty()) {
             serviceNames.add(context.getRemoteServiceName());
           }
 
           List<Metric> actualMetricList = Lists.newArrayList();
-          addMetrics(
-              CloudWatchService.SERVICE_DIMENSION,
-              serviceNames,
-              expectedMetricList,
-              actualMetricList);
-          addMetrics(
-              CloudWatchService.REMOTE_SERVICE_DIMENSION,
-              remoteServiceNames,
-              expectedMetricList,
-              actualMetricList);
+
+          // Add sets of dimesion filters to use for each query to CloudWatch.
+          List<List<Pair<String, String>>> dimensionLists = Lists.newArrayList();
+          // Query metrics that includes any of these <service> values.
+          for (String serviceName : serviceNames) {
+            dimensionLists.add(
+                Arrays.asList(new Pair<>(CloudWatchService.SERVICE_DIMENSION, serviceName)));
+          }
+          // Query metrics that includes any of these <remoteService> values.
+          for (String remoteServiceName : remoteServiceNames) {
+            dimensionLists.add(
+                Arrays.asList(
+                    new Pair<>(CloudWatchService.REMOTE_SERVICE_DIMENSION, remoteServiceName)));
+          }
+          // Query for metrics that includes both of these <remoteService, remoteTarget> values.
+          // Querying just 'remoteService="AWS.SDK.S3"' would also work, but that can result in
+          // returning too many canary metrics and may cause issues.
+          if (context.getRemoteTargetName() != null && !context.getRemoteTargetName().isEmpty()) {
+            dimensionLists.add(
+                Arrays.asList(
+                    new Pair<>(CloudWatchService.REMOTE_SERVICE_DIMENSION, "AWS.SDK.S3"),
+                    new Pair<>(CloudWatchService.REMOTE_TARGET_DIMENSION, context.getRemoteTargetName())));
+          }
+
+          // Populate actualMetricList with metrics that pass through at least one of the dimension filters
+          for (List<Pair<String, String>> dimensionList : dimensionLists) {
+            addMetrics(dimensionList, expectedMetricList, actualMetricList);
+          }
```

Comment:
Have you double checked that this doesn't pull any unwanted metrics? I can help double check this if you have.

Reply:
From looking at the code perspective, all above queries include `${{ env.TESTING_ID }}` or `${{ env.REMOTE_SERVICE_IP }}`, so that provides some confidence that unwanted metrics are not included.

https://github.com/jj22ee/aws-application-signals-test-framework/actions/runs/8913586327/job/24479367762
Otherwise, a quick check in my tests shows that the `actual metrics` list is:
- about [40,000 characters long](https://github.com/jj22ee/aws-application-signals-test-framework/actions/runs/8913586327/job/24479367762) compared to [current sample 36,000 characters](https://github.com/aws-observability/aws-application-signals-test-framework/actions/runs/8913715520/job/24479785297) long, likely due to the added queries in this PR
- Only one test id `"-629"` is referenced in the actual metrics list